package/kedro_viz/server.py

```diff
@@ -317,90 +330,143 @@ def _construct_layer_mapping(catalog):
+    name = name.replace("-", " ").replace("_", " ")
+    parts = [n[0].upper() + n[1:] for n in name.split()]
+    return " ".join(parts)
+
+
+def format_pipelines_data(pipelines: Dict[str, "Pipeline"], catalog) -> Dict[str, list]:
     """
-    Format pipeline and catalog data from Kedro for kedro-viz
+    Format pipelines and catalog data from Kedro for kedro-viz.
 
     Args:
-        pipeline: Kedro pipeline object
-        catalog:  Kedro catalog object
-    """
+        pipelines: Dictionary of Kedro pipeline objects.
+        catalog: Kedro catalog object.
 
-    def pretty_name(name):
-        name = name.replace("-", " ").replace("_", " ")
-        parts = [n[0].upper() + n[1:] for n in name.split()]
-        return " ".join(parts)
+    Returns:
+        Dictionary of pipelines, nodes, edges, tags and layers, and pipelines list.
 
+    """
+    pipeline_list = []
+    # keep track of a sorted list of nodes to returned to the client
+    node_list = []
+    # keep track of edges in the graph: [{source_node_id -> target_node_id}]
+    edge_list = []
     # keep tracking of node_id -> node data in the graph
     nodes = {}
-    # keep track of a sorted list of nodes to returned to the client
-    sorted_nodes = []
     # keep track of node_id -> set(child_node_ids) for layers sorting
     node_dependencies = defaultdict(set)
-    # keep track of edges in the graph: [{source_node_id -> target_node_id}]
-    edges = []
+    tags = set()
+
+    for pipeline_key, pipeline in pipelines.items():
+        pipeline_list.append({"id": pipeline_key, "name": _pretty_name(pipeline_key)})
+        format_pipeline_data(
+            pipeline_key,
+            pipeline,
+            catalog,
+            nodes,
+            node_dependencies,
+            tags,
+            edge_list,
+            node_list,
+        )
+
+    # sort tags
+    sorted_tags = [{"id": tag, "name": _pretty_name(tag)} for tag in sorted(tags)]
+    # sort layers
+    sorted_layers = _sort_layers(nodes, node_dependencies)
+    return {
+        "nodes": node_list,
+        "edges": edge_list,
+        "tags": sorted_tags,
+        "layers": sorted_layers,
+        "pipelines": pipeline_list,
+    }
+
+
+# pylint: disable=too-many-locals,too-many-arguments
+def format_pipeline_data(
+    pipeline_key: str,
+    pipeline: "Pipeline",  # noqa: F821
+    catalog: "DataCatalog",
+    nodes: Dict[str, dict],
+    node_dependencies: Dict[str, dict],
+    tags: Set[str],
+    edge_list: List[dict],
+    node_list: List[dict],
+) -> None:
+    """Format pipeline and catalog data from Kedro for kedro-viz.
+
+    Args:
+        pipeline_key: key value of a pipeline object (e.g "__default__").
+        pipeline: Kedro pipeline object.
+        catalog:  Kedro catalog object.
+        nodes: Dictionary of id and node dict.
+        node_dependencies: Dictionary of id and node dependencies.
+        edge_list: List of all edges.
+        node_list: List of all nodes.
```

Comment:
```suggestion
        node_list: List of the pipeline nodes.
```
These are nodes related to the specific pipeline.

Reply:
Actually it's a list of all nodes for all pipelines (same for edges)